includes/class-cli.php

```diff
@@ -23,15 +25,88 @@ class Cli extends \WP_CLI_Command {
 	 *
 	 * ## EXAMPLES
 	 *
-	 *     $ wp activitypub self-destruct
+	 *     $ wp activitypub self_destruct
 	 *
 	 * @param array|null $args       The arguments.
 	 * @param array|null $assoc_args The associative arguments.
 	 *
 	 * @return void
 	 */
-	public function self_destruct( $args, $assoc_args ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		\WP_CLI::warning( 'Self-Destructing is not implemented yet.' );
+	public function self_destruct( $args, $assoc_args = array() ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		// Display warning header.
+		\WP_CLI::line( \WP_CLI::colorize( '%R⚠️  DESTRUCTIVE OPERATION ⚠️%n' ) );
+		\WP_CLI::line( '' );
+
+		$question = 'You are about to delete your blog from the Fediverse. This action is IRREVERSIBLE and will:';
+		\WP_CLI::line( \WP_CLI::colorize( "%y{$question}%n" ) );
+		\WP_CLI::line( \WP_CLI::colorize( '%y• Send Delete activities to all followers%n' ) );
+		\WP_CLI::line( \WP_CLI::colorize( '%y• Remove your blog from ActivityPub networks%n' ) );
+		\WP_CLI::line( '' );
+
+		\WP_CLI::confirm( 'Are you absolutely sure you want to continue?', $assoc_args );
+
+		// Get all users with ActivityPub capabilities.
+		$user_ids = \get_users(
+			array(
+				'fields'         => 'ID',
+				'capability__in' => array( 'activitypub' ),
+			)
+		);
+
+		if ( empty( $user_ids ) ) {
+			\WP_CLI::warning( 'No ActivityPub users found. Nothing to delete.' );
+			return;
+		}
+
+		// Show what will be processed.
+		$user_count = \count( $user_ids );
+		\WP_CLI::line( \WP_CLI::colorize( '%GStarting Fediverse deletion process...%n' ) );
+		\WP_CLI::line( \WP_CLI::colorize( "%BFound {$user_count} ActivityPub user(s) to process:%n" ) );
+		\WP_CLI::line( '' );
+
+		// Set the self-destruct flag.
+		\add_option( 'activitypub_self_destruct', true );
+
+		// Hook into outbox processing completion to notify when self-destruct is done.
+		\add_action( 'activitypub_outbox_processing_complete', array( Scheduler::class, 'check_self_destruct_completion' ), 10, 2 );
```

Comment:
Does this work? Any outbox processing should happen in separate threads where this action would not be added.

Reply:
good question!

Reply:
This felt like the best UX. We should get sure that this delete is properly processed, because otherwise this could end up in a big inconsistancy that can never be properly cleaned up!?